### PR TITLE
Add Java Action Permissions

### LINKED ISSUE #952: Implement permission checking for merlin java actions
### Checked for duplicates

No - I haven't checked

### Alternatives considered

No - I haven't considered

### Related problems

I'm frustrated when [...] happens as documented in issue-XYZ

### Describe the feature request

I need or want [...]

### LINKED ISSUE #953: Implement permission checking for scheduler java actions

### PR DESCRIPTION
**Pull request by Mythicaeda**
* **Tickets addressed:** Closes #953. Closes #952.
* **Review:** By commit  
* **Merge strategy:** Merge (no squash)  

## Description

- Moved the Permissions Service to its own module. It now uses GraphQL queries
- Add permissions checking for the `schedule` action

## Verification

- E2E Tests have been updated to allow Hasura headers to be specified. By default, the GQL queries are made using the `aerie_admin` role and the `Aerie Legacy` username

## Documentation

[Docs ticket](https://github.com/NASA-AMMOS/aerie-docs/issues/61)

## Future work

Further refactors to pull shared code into shared modules. Highlight: as of this PR, we now have _3_ `PlanId` classes, all of which have the same implementation (a record containing a `long id`).

### CODE REVIEW

File: permissions/src/main/java/gov/nasa/jpl/aerie/permissions/exceptions/Unauthorized.java [hunk 1]
```diff
@@ -0,0 +1,25 @@
+package gov.nasa.jpl.aerie.permissions.exceptions;
+
+import gov.nasa.jpl.aerie.permissions.Action;
+import gov.nasa.jpl.aerie.permissions.PermissionType;
+import gov.nasa.jpl.aerie.permissions.gql.PlanId;
+
+public class Unauthorized extends Exception {
+  public Unauthorized(final String role, final Action action) {
+    super("Role '" + role + "' is not allowed to perform action '" + action + "'");
```
Comment (skovati) [inline permissions/src/main/java/gov/nasa/jpl/aerie/permissions/exceptions/Unauthorized.java]: Very minor, but we could replace all the strings in these super calls with text blocks and `formatted` as a standard.

```suggestion
    super("""Role "%s" is not allowed to perform action "%s".""".formatted(role, action));
```

File: permissions/src/main/java/gov/nasa/jpl/aerie/permissions/gql/PlanId.java
Comment (skovati) [inline permissions/src/main/java/gov/nasa/jpl/aerie/permissions/gql/PlanId.java:1]: Don't we already have `merlin-server/src/main/java/gov/nasa/jpl/aerie/merlin/server/models/PlanId.java`?

File: settings.gradle [hunk 2]
```diff
@@ -10,6 +10,7 @@ include 'contrib'
 
 // Service support
 include 'parsing-utilities'
+include 'permissions'
```
Comment (skovati) [inline settings.gradle:13]: Can we add this new top level module description into our [README](https://github.com/NASA-AMMOS/aerie#directory-structure)?

**Review by skovati — APPROVED** [review]
This looks fantastic to me! Thanks for fixing up my comments on @mattdailis's original PR. The new permission module is 🔥. Just have a few tiny non-blocking comments.

File: permissions/src/main/java/gov/nasa/jpl/aerie/permissions/gql/PlanId.java
Comment (Mythicaeda) [reply permissions/src/main/java/gov/nasa/jpl/aerie/permissions/gql/PlanId.java:1]: We do, but if we were to use it, we'd create a circular dependency between the `merlin-server` and `permissions` modules

File: permissions/src/main/java/gov/nasa/jpl/aerie/permissions/exceptions/Unauthorized.java (hunk 1 above)
Comment (Mythicaeda) [reply permissions/src/main/java/gov/nasa/jpl/aerie/permissions/exceptions/Unauthorized.java]: Done!

File: settings.gradle (hunk 2 above)
Comment (Mythicaeda) [reply settings.gradle:13]: Done!

**Comment by skovati** [comment]
🚀

**Pull request merged**